java/ql/src/experimental/semmle/code/java/security/StaticInitializationVectorQuery.qll

```diff
@@ -0,0 +1,166 @@
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.TaintTracking2
+
+/**
+ * Holds if `array` is initialized only with constants, for example,
+ * `new byte[8]` or `new byte[] { 1, 2, 3, 4, 5, 6, 7, 8 }`.
+ */
+private predicate initializedWithConstants(ArrayCreationExpr array) {
+  not exists(array.getInit())
+  or
+  forex(Expr element | element = array.getInit().getAChildExpr() |
```

Comment:
Is there a reason why you chose `getAChildExpr()` instead of [`getAnInit()`](https://codeql.github.com/codeql-standard-libraries/java/semmle/code/java/Expr.qll/predicate.Expr$ArrayInit$getAnInit.0.html)? (though both are probably functionally equivalent)

Reply:
No, I don't have a reason. As far as I remember, I borrowed it [here](https://github.com/github/codeql/blob/ef0ea247c40da805efa427a37f5213457d18f714/java/ql/src/Security/CWE/CWE-798/HardcodedCredentials.qll#L12). I suppose `getAnInit()` should work fine as well.

Reply:
`getAnInit()` is indeed better

Reply:
Okay, updated.